src/execution_plan/ops/op_update.c

```diff
@@ -137,12 +136,12 @@ void _UpdateSchemas(const OpUpdate *op) {
         /* Locate node label. */
         AST_GraphEntity* ge = MatchClause_GetEntity(op->ast->matchNode, entityAlias);
         char *l = ge->label;
-        LabelStoreType t = (ge->t == N_ENTITY) ? STORE_NODE : STORE_EDGE;
-
-        LabelStore *store = LabelStore_Get(op->ctx, t, op->graphName, l);
+        LabelStore *store = GraphContext_GetNodeStore(op->gc, l);
+        // TODO we don't update allStore if label store isn't found?
```

Comment:
I don't think it is possible at this point that label store is not found, as we're updating an existing entity, as such the entity must be labeled as l.

Reply:
I don't think that's safe - we are updating existing entities, but retrieving the label from the AST match clause, which is not required to be provided:
`redis-cli GRAPH.QUERY imdb "MATCH (a) SET a.universal_prop = 10"`
```
137             AST_GraphEntity* ge = MatchClause_GetEntity(op->ast->matchNode, entityAlias);
(gdb)
138             char *l = ge->label;
(gdb) p *ge
$1 = {
  alias = 0x557069f76200 "a",
  label = 0x0,
  properties = 0x0,
  t = N_ENTITY
```
This is admittedly a kind of silly operation, but still a valid one - in fact, the appropriate label schemas need to be updated in addition to the all store. I think I have some code in the index update PR that deals with this, though.

Reply:
This still needs to be addressed, though I'd rather do it in a separate PR after we discuss approaches.

Reply:
I think we should see how Neo is handling this, i.e. updating an entity, where a label isn't specified, do they fetch all labels applied to that entity and perform the necessary updates?